packages/design-system/src/components/input-field.tsx

```diff
@@ -175,24 +177,38 @@ export const InputField = forwardRef(
       suffix,
       containerRef,
       inputRef,
+      onFocus,
+      onBlur,
       ...props
     }: InputProps & {
       prefix?: ReactNode;
       suffix?: ReactNode;
       containerRef?: Ref<HTMLDivElement>;
       inputRef?: Ref<HTMLInputElement>;
+      onFocus?: (e: FocusEvent) => void;
+      onBlur?: (e: FocusEvent) => void;
     },
     ref: Ref<HTMLDivElement>
-  ) => (
-    <Container
-      css={css}
-      className={className}
-      prefix={prefix}
-      suffix={suffix}
-      ref={mergeRefs(ref, containerRef ?? null)}
-    >
-      <Input {...props} ref={inputRef} />
-    </Container>
-  )
+  ) => {
+    // Our input field can contain multiple focused elements,
+    // so we need to use useFocusWithin to track focus within the container.
+    const { focusWithinProps } = useFocusWithin({
+      onFocusWithin: (e) => onFocus?.(e),
```

Comment:
why do you need this wrapping instead of this?
```suggestion
      onFocusWithin: onFocus,
```